client/core/core.go

```diff
@@ -3559,6 +3560,32 @@ func (c *Core) Login(pw []byte) (*LoginResult, error) {
 	}
 	defer crypter.Close()
 
+	loginResult := func() *LoginResult {
+		dexStats := c.initializeDEXConnections(crypter)
```

Comment:
> Is there a use case for having multiple Login calls all blocking each other and effectively queuing up, as currently proposed?

When users `Login` for the first time, It can take a considerable amount of time (maybe because of SPV wallet failing to connect to suitable peers at the time). Users can become impatient, refresh, and send another login request. 

This suggestion will effectively queue subsequent requests, wait for the initial Login if any and return a response.

Reply:
> An alternate approach I probably would have taken would be to immediately return an error from Login if another Login were already running.

The frontend depends on the `Login` response to load the wallets page. If for any reason, the user refreshes and gets an error because another `Login` is already running or the user is already Logged in, the frontend will be stuck on the login page.

Reply:
I see.  User gets impatient, reloads page, tries to login again, so we want to give them a response without actually processing the request concurrent with the previous.  Returning an error will leave the user in a state on the UI where they just have to retry until it works -- bad UX.  On the other hand, hanging again on the subsequent requests will lead them to retry several again and again.

I have to think about the `initializeDEXConnections` issue then.  If the user does as described, then when the first one unblocks and the other queued requests get handled, it will do `c.authDEX` in rapid succession for however many re-request they made while they were sitting there impatiently.

Reply:
Another issue arises if there's more than one `Login` request waiting (three or more concurrent requests).  Say there are three: 1 doing work, and 2 sitting at `c.loginWg.Wait()`.  When that statement unblocks, both of those two will `initializeDEXConnections` simultaneously.  Concurrent login `authDEX` calls.  That's not desirable.

Reply:
Ah, the `if dc.acct.authed() {` condition in `initializeDEXConnections` should prevent repeated calls to `authDEX`.  Maybe this is OK.
Beginning deep dive and testing...

Reply:
>  When that statement unblocks, both of those two will initializeDEXConnections simultaneously. Concurrent login authDEX calls. That's not desirable.

Hmm, there should be a way around this.

Reply:
> When users Login for the first time, It can take a considerable amount of time (maybe because of SPV wallet failing to connect to suitable peers at the time).

This seems to be an issue that needs solving on its own. Finding peers should not slow down log in correct?

Reply:
> Finding peers should not slow down log in correct?

That doesn't.  It starts the sync loop and login is done regardless.

Reply:
> That doesn't. It starts the sync loop and login is done regardless.

I've been stuck on the login page for a while some time ago. I noticed btc spv wallet took a huge amount of time to get ready. It got so bad that I had to open and see what was going on in `neutrino.log`(I don't have the logs anymore). It turned out btc spv was trying to connect to a banned peer and always rejecting it until it found a non-banned peer.

Reply:
It looks like we wait for all the `connectAndUpdateWallet` to finish when logging in. Which seems to be a long time sometimes for internal wallets that have not been finding peers/syncing beforehand.